src/integrator/cuda/cuda_weights.cu

```diff
@@ -14,33 +14,59 @@ namespace cuda       {
 
 using namespace GauXC::cuda;
 
+__global__ void reciprocal_kernel(size_t length, double* vec) {
+   for (int i = threadIdx.x + blockIdx.x * blockDim.x; i < length; i += blockDim.x * gridDim.x) {
+     vec[i] = 1. / vec[i];
+   }
+}
 
 __global__ void compute_point_center_dist(
         size_t      npts,
+        size_t      LDatoms,
         size_t      natoms,
   const double*     coords,
   const double*     points,
         double*     dist
 ) {
 
+  __shared__ double3 point_buffer[32];
+  register double3 coord_reg;
 
-  const int tid_x = threadIdx.x + blockIdx.x * blockDim.x;
-  const int tid_y = threadIdx.y + blockIdx.y * blockDim.y;
-
-
-  if( tid_y < natoms and tid_x < npts ) {
-
-    const int iAtom = tid_y;
-    const int iPt   = tid_x;
+  const int natoms_block = (natoms + 31) / 32;
```

Comment:
I would calculate these relative to the `warp_size` if possible

Reply:
Do we know if `nvcc` is smart enough to desipher / optimize around a `constexpr` constant = 32 vs just manually specifying `32` everywhere? My hope is yes, but I'm not sure I've ever confirmed this

Reply:
I confirmed via the generated PTX that in this case `nvcc` is able to decipher the `constexpr` and optimize around it.

Reply:
Good to know for future reference!